src/client/test/utils.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { StaticRouter } from 'react-router'
+import { render, RenderOptions } from '@testing-library/react'
+import { renderHook, RenderHookOptions } from '@testing-library/react-hooks'
+
+interface AllProvidersProps {
+  location?: string
+}
+
+const AllProviders: React.FC<AllProvidersProps> = ({ children, location }) => {
+  return <StaticRouter location={location}>{children}</StaticRouter>
+}
+
+const customRender = (
+  ui: React.ReactElement,
+  options?: Omit<RenderOptions, 'wrapper'> & {
+    wrapperProps?: Record<string, any>
+  },
+) =>
+  render(ui, {
+    wrapper: (props: any) => (
+      <AllProviders {...props} {...options?.wrapperProps} />
```

Comment:
Right, so the `wrapperProps` are provided by the user when calling `customRender`. The other `props` are provided by testing-library itself when it eventually calls the `wrapper` function. It will pass things like `children`. That's why we need both!

Code reference here: https://github.com/testing-library/react-testing-library/issues/780

Reply:
🆒 👍